MacroPad_Scramble_Lock/code.py

```diff
@@ -0,0 +1,212 @@
+# axctivate a solenoid which will unlock, like a door
+solenoid = DigitalInOut(board.SDA)
+solenoid.direction = Direction.OUTPUT
+
+# FUNCTIONS ------------------
+
+def keys_clear():  # Set display in the Start mode, key LEDs off
+    for i in range(12):
+        macropad.pixels[i] = 0x000000
+        group[i].text = " "
+    macropad.pixels.show()
+    group[9].text = "START"
+    macropad.display.show(group)
+    macropad.display.refresh()
+
+def scramble():  # Scramble values of the keys and display on screen
+    for times in range(5):
+        # The following lines implement a random.shuffle method
+        # See https://www.rosettacode.org/wiki/Knuth_shuffle#Python
+        # random.shuffle(key_values)           # Shuffle the key array
+        for i in range(len(key_values)-1, 0, -1):
+            j = random.randrange(i + 1)
+            key_values[i], key_values[j] = key_values[j], key_values[i]
+        keys_display()
+        macropad.play_tone(tones[times], 0.3)  # Play a tone each scramble iteration
+        time.sleep(0.01)
+
+def keys_display():   # Display the current values of the keys on screen
+    for k in range(9):  # The first 9 keys
+        group[k].text = str(key_values[k])
+        macropad.pixels[k] = colors[key_values[k]]
+    group[10].text = str(key_values[9])  # The 'Zero' position number
+    group[9].text = " "   # Start blanks
+    group[11].text = " "  # Status blanks
+    macropad.pixels[10] = colors[key_values[9]]
+    macropad.display.refresh()
+    macropad.pixels.show()
+
+# INITIALIZATION -----------------------
+
+macropad = MacroPad()  # Set up MacroPad library and behavior
+macropad.display.auto_refresh = False
+macropad.pixels.auto_write = False
+
+# Set up displayio group with all the labels
+group = displayio.Group()
+for key_index in range(12):
+    x = key_index % 3
+    y = key_index // 3
+    group.append(label.Label(terminalio.FONT, text='', color=0xFFFFFF,
+                             anchored_position=((macropad.display.width - 1) * x / 2,
+                                                macropad.display.height - 1 -
+                                                (3 - y) * 12),
+                             anchor_point=(x / 2, 1.0)))
+group.append(Rect(0, 0, macropad.display.width, 12, fill=0xFFFFFF))
+group.append(label.Label(terminalio.FONT, text='ScramblePad', color=0x000000,
+                         anchored_position=(macropad.display.width//2, -2),
+                         anchor_point=(0.5, 0.0)))
+
+
+# Initialize in a clear state
+state = STATE_CLEAR
+macropad.keyboard.release_all()
+keys_clear()
+solenoid.value = False
+
+# MAIN LOOP ----------------------------
+
+while True:
+    if state == STATE_RESET:
+        print("Reset state")
+        macropad.keyboard.release_all()
+        password_guess = ""  # Reset password entry
+        state = STATE_CLEAR  # Reset state
+
+    # Check for key presses/releases
+    event = macropad.keys.events.get()
```

Comment:
For future reference, we prefer to call this variable `key_event` because it's clearer.

Reply:
Thanks, copypasta from Drag0n's code, noted